### DISCUSSION TITLE
Confused document

### DISCUSSION THREAD

**Discussion by leapit** [ory/hydra#2740] [Q&A] [answered]
The REST api doc [https://www.ory.sh/hydra/docs/next/reference/api/#operation/oauth2Token](https://www.ory.sh/hydra/docs/next/reference/api/#operation/oauth2Token) hint
```
Do not implement a client for this endpoint yourself. Use a library. There are many libraries available for any programming language. You can find a list of libraries here: https://oauth.net/code/

Do note that Hydra SDK does not implement this endpoint properly. Use one of the libraries listed above!
```
Confused by above `Hydra SDK does not implement this endpoint properly`,does that mean the project https://github.com/ory/hydra-login-consent-node is not for production,need implement myself?

and also confused by the parameters
```
grant_type
code	
refresh_token	
redirect_uri	
client_id
```
what does `code` stand for?there is no explanation.

sad for document.

**Comment by Benehiko** [comment, marked as answer]
Hi @leapit 

It is not advised to directly use the endpoints, you need to use a library that implements good security practices. The `login-consent-node` application is an example and should not be directly copied for production usage.

As for the parameters, maybe you should first read this section of the docs: https://www.ory.sh/hydra/docs/next/concepts/openid-connect-oidc
